### ISSUE TITLE
BUG: Azure E2e workflow targeting Azure env w/ an AKS cluster hosting Harbor fails to complete successfully

### ISSUE THREAD

**Issue by pacphi** [pacphi/gha-workflows-with-gitops-for-tanzu-application-platform#16]
An OSS Helm chart is used to install Harbor.

Here's the [values.yml](https://github.com/pacphi/gha-workflows-with-gitops-for-tanzu-application-platform/blob/main/terraform/k8s/harbor/templates/values.tpl) template

`harbor-database-0` pod fails to start with

```bash
❯ k logs harbor-database-0 -n harbor
Defaulted container "database" out of: database, data-migrator (init), data-permissions-ensurer (init)
init DB, DB version:13
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locales
  COLLATE:  en_US.UTF-8
  CTYPE:    en_US.UTF-8
  MESSAGES: C
  MONETARY: C
  NUMERIC:  C
  TIME:     C
The default text search configuration will be set to "english".

Data page checksums are disabled.

creating directory /var/lib/postgresql/data/pgdata/pg13 ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... UTC
creating configuration files ... ok
2023-03-18 00:02:25.526 UTC [22] LOG:  could not link file "pg_wal/xlogtemp.22" to "pg_wal/000000010000000000000001": Unknown error 524
2023-03-18 00:02:25.543 UTC [22] FATAL:  could not open file "pg_wal/000000010000000000000001": No such file or directory
child process exited with exit code 1
initdb: removing data directory "/var/lib/postgresql/data/pgdata/pg13"
running bootstrap script ... %                                                                                                                                                                                                                                                                      
❯ k logs harbor-database-0 -n harbor
Defaulted container "database" out of: database, data-migrator (init), data-permissions-ensurer (init)
init DB, DB version:13
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locales
  COLLATE:  en_US.UTF-8
  CTYPE:    en_US.UTF-8
  MESSAGES: C
  MONETARY: C
  NUMERIC:  C
  TIME:     C
The default text search configuration will be set to "english".

Data page checksums are disabled.

creating directory /var/lib/postgresql/data/pgdata/pg13 ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... UTC
creating configuration files ... ok
2023-03-18 00:02:25.526 UTC [22] LOG:  could not link file "pg_wal/xlogtemp.22" to "pg_wal/000000010000000000000001": Unknown error 524
2023-03-18 00:02:25.543 UTC [22] FATAL:  could not open file "pg_wal/000000010000000000000001": No such file or directory
child process exited with exit code 1
initdb: removing data directory "/var/lib/postgresql/data/pgdata/pg13"
running bootstrap script ...
```

Not sure what to do here.

Using this storage controller `azureblob-nfs-premium`, see

```bash
❯ k get sc -A
NAME                              PROVISIONER          RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
azureblob-fuse-premium            blob.csi.azure.com   Delete          Immediate              true                   24m
azureblob-nfs-premium (default)   blob.csi.azure.com   Delete          Immediate              true                   24m
azurefile                         file.csi.azure.com   Delete          Immediate              true                   24m
azurefile-csi                     file.csi.azure.com   Delete          Immediate              true                   24m
azurefile-csi-premium             file.csi.azure.com   Delete          Immediate              true                   24m
azurefile-premium                 file.csi.azure.com   Delete          Immediate              true                   24m
default                           disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   24m
managed                           disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   24m
managed-csi                       disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   24m
managed-csi-premium               disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   24m
managed-premium                   disk.csi.azure.com   Delete          WaitForFirstConsumer   true                   24m
```

Here are the persistent volume claims and persistent volumes.  Describing them does not offer anything appreciably useful.  They are in good shape.

```bash
❯ k get pv,pvc -A
NAME                                                        CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                                    STORAGECLASS            REASON   AGE
persistentvolume/pvc-148ef847-3be1-4242-b3c6-c1780fc658b2   10Gi       RWO            Delete           Bound    harbor/data-harbor-trivy-0               azureblob-nfs-premium            16m
persistentvolume/pvc-2f57981a-7492-4a0b-be5c-fac4394b78cd   100Gi      RWO            Delete           Bound    harbor/harbor-registry                   azureblob-nfs-premium            16m
persistentvolume/pvc-463006f5-7dd6-43e3-9f14-610fe5539cba   2Gi        RWO            Delete           Bound    harbor/data-harbor-redis-0               azureblob-nfs-premium            16m
persistentvolume/pvc-8b4ae319-6aeb-4a8a-8fa2-90816a43148f   5Gi        RWO            Delete           Bound    harbor/harbor-jobservice                 azureblob-nfs-premium            16m
persistentvolume/pvc-a719d483-6804-4900-b9f5-35c89d1a0ebd   20Gi       RWO            Delete           Bound    harbor/harbor-chartmuseum                azureblob-nfs-premium            16m
persistentvolume/pvc-e20aed58-40e3-4b45-a9b7-f70b81c064bb   1Gi        RWO            Delete           Bound    harbor/database-data-harbor-database-0   azureblob-nfs-premium            16m

NAMESPACE   NAME                                                    STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS            AGE
harbor      persistentvolumeclaim/data-harbor-redis-0               Bound    pvc-463006f5-7dd6-43e3-9f14-610fe5539cba   2Gi        RWO            azureblob-nfs-premium   16m
harbor      persistentvolumeclaim/data-harbor-trivy-0               Bound    pvc-148ef847-3be1-4242-b3c6-c1780fc658b2   10Gi       RWO            azureblob-nfs-premium   16m
harbor      persistentvolumeclaim/database-data-harbor-database-0   Bound    pvc-e20aed58-40e3-4b45-a9b7-f70b81c064bb   1Gi        RWO            azureblob-nfs-premium   16m
harbor      persistentvolumeclaim/harbor-chartmuseum                Bound    pvc-a719d483-6804-4900-b9f5-35c89d1a0ebd   20Gi       RWO            azureblob-nfs-premium   16m
harbor      persistentvolumeclaim/harbor-jobservice                 Bound    pvc-8b4ae319-6aeb-4a8a-8fa2-90816a43148f   5Gi        RWO            azureblob-nfs-premium   16m
harbor      persistentvolumeclaim/harbor-registry                   Bound    pvc-2f57981a-7492-4a0b-be5c-fac4394b78cd   100Gi      RWO            azureblob-nfs-premium   16m
```

**Comment by pacphi** [comment]
As it turns out the `metadata-store` component of TAP also relies on a Postgres db that requires a PVC, and we're failing in a similar manner. Something is awry.

```bash
❯ k logs metadata-store-db-0 -n metadata-store
Defaulted container "postgres" out of: postgres, change-cert-ownership (init)
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.utf8".
The default database encoding has accordingly been set to "UTF8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data/pgdata ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... UTC
creating configuration files ... ok
2023-03-28 02:25:39.517 UTC [26] LOG:  could not link file "pg_wal/xlogtemp.26" to "pg_wal/000000010000000000000001": Unknown error 524
2023-03-28 02:25:39.528 UTC [26] FATAL:  could not open file "pg_wal/000000010000000000000001": No such file or directory
child process exited with exit code 1
initdb: removing contents of data directory "/var/lib/postgresql/data/pgdata"
running bootstrap script ...
```

**Comment by pacphi** [comment]
Fixed in main (1.4.2) and in 1.5.0 branch.
